plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/expression/ExpressionConverter.java

```diff
@@ -0,0 +1,260 @@
+        }
+        return new Literal<>(literal, constantType);
+    }
+
+    private Expression convertTrinoCall(Call trinoCall)
+    {
+        FunctionName functionName = trinoCall.getFunctionName();
+        List<Expression> convertedArguments = new ArrayList<>();
+        for (ConnectorExpression arg : trinoCall.getArguments()) {
+            convertedArguments.add(convertExpressionRecursive(arg));
+        }
+
+        return mapTrinoFunctionToCustomExpression(functionName, convertedArguments, trinoCall.getType());
+    }
+
+    private Expression mapTrinoFunctionToCustomExpression(
+            FunctionName functionName, List<Expression> arguments, io.trino.spi.type.Type trinoType)
+    {
+        Type hudiReturnType = HudiTypeConverter.getType(trinoType);
+
+        Expression finalHudiExpression;
+
+        // Logical Operators
+        if (StandardFunctions.AND_FUNCTION_NAME.equals(functionName)) {
+            if (arguments.isEmpty()) {
+                // Handle case of empty arguments for AND if necessary
+                throw new IllegalArgumentException("AND function expects at least one argument.");
+            }
+            if (arguments.size() == 1) {
+                // AND(X) is just X
+                finalHudiExpression = arguments.getFirst();
+            }
+            // Iteratively build the AND chain
+            Expression result = arguments.get(0);
+            for (int i = 1; i < arguments.size(); i++) {
+                result = Predicates.and(result, arguments.get(i));
+            }
+            finalHudiExpression = result;
+        }
+        else if (StandardFunctions.OR_FUNCTION_NAME.equals(functionName)) {
+            if (arguments.isEmpty()) {
+                // Handle case of empty arguments for OR if necessary
+                throw new IllegalArgumentException("OR function expects at least one argument.");
+            }
+            if (arguments.size() == 1) {
+                // OR(X) is just X
+                return arguments.getFirst();
+            }
+            // Iteratively build the OR chain
+            Expression result = arguments.getFirst();
+            for (int i = 1; i < arguments.size(); i++) {
+                result = Predicates.or(result, arguments.get(i));
+            }
+            finalHudiExpression = result;
+        }
+        else if (StandardFunctions.NOT_FUNCTION_NAME.equals(functionName) && arguments.size() == 1) {
+            finalHudiExpression = Predicates.not(arguments.getFirst());
+        }
+        // Comparison Operators
+        else if (StandardFunctions.EQUAL_OPERATOR_FUNCTION_NAME.equals(functionName) && arguments.size() == 2) {
+            finalHudiExpression = Predicates.eq(arguments.get(0), arguments.get(1));
+        }
+        else if (StandardFunctions.NOT_EQUAL_OPERATOR_FUNCTION_NAME.equals(functionName) && arguments.size() == 2) {
+            finalHudiExpression = Predicates.not(Predicates.eq(arguments.get(0), arguments.get(1)));
```

Comment:
Is there any inefficiency of having two levels of expression during predicate evaluation, instead of sth like a `Predicates.neq`?

Reply:
This should be negligible as it will only be evaluated at most once. This is written the way it is as `Predicates` belongs to Hudi project scope: `org.apache.hudi.expression.Predicates` and does not have the `neq` function.